src/top/utop.el

```diff
@@ -488,12 +488,12 @@ it is started."
   "Insert the phrase terminator at the end of buffer."
   ;; Search the longest suffix of the input which is a prefix of the
   ;; phrase terminator
-  (let* ((end (point-max))
-         (pos (max utop-prompt-max (- end (length utop-phrase-terminator)))))
-    (while (not (string-prefix-p (buffer-substring-no-properties pos end) utop-phrase-terminator))
-      (setq pos (1+ pos)))
+  (let ((end (point-max)))
+    (goto-char (max utop-prompt-max (- end (length utop-phrase-terminator))))
+    (while (not (string-prefix-p (buffer-substring-no-properties (point) end) utop-phrase-terminator))
```

Comment:
Frankly, this feel super wasteful. We're moving char by char to check if the last two characters are `;;`? Seems to me it'd be much easier to check the suffix of the string and insert the `;;` if not present. You can use either `string-suffix-p`, regex directly or `looking-back`. This would simplify this convoluted code a lot IMO.

Reply:
I agree it's wasteful, but I was trying to preserve the semantics of the old code, which attempted to account for the case where the phrase terminator was something else.

From a quick glance at the OCaml files, the phrase terminator can be a single semicolon (https://github.com/diml/utop/blob/c68acd1ff3c350f33ef8c270fa3de236f832f5e1/src/lib/uTop.mli#L104-L106) when using "revised" syntax. However, grepping for `";"` turns up a total of two results, both of which look irrelevant. I'm not very knowledgeable with OCaml, but it looks like revised syntax isn't encouraged. Are there any other values that the phrase terminator can take? Should we drop any mentions of support for revised syntax and always require two semicolons?

Reply:
If I'm not mistaken, the revised syntax was taken from Camlp4/Camlp5, can you confirm? If so, then utop has dropped support for it already, which means that two semicolons should indeed be safe.

Reply:
I guess @rgrinberg will know best, but I've never seen anything other than `;;` myself.

Reply:
We don't need to support the revised syntax at all. It's been dead for a long time already.